tests/publiccloud/migration.pm

```diff
@@ -21,42 +21,117 @@ use utils;
 use publiccloud::utils;
 
 our $target_version = get_required_var('TARGET_VERSION');
-our $not_clean_vm = get_var('PUBLIC_CLOUD_NO_CLEANUP_ON_FAILURE');
+our $not_clean_vm   = get_var('PUBLIC_CLOUD_NO_CLEANUP_ON_FAILURE');
 
 sub run {
     my ($self, $args) = @_;
     select_serial_terminal();
     my $provider = $args->{my_provider};
-    my $instance = $provider->create_instance();
-    $instance->wait_for_guestregister();
+    my $instance = $provider->create_instance(check_guestregister => 0);
     registercloudguest($instance) if is_byos();
 
     register_addons_in_pc($instance);
-    $instance->ssh_assert_script_run('sudo zypper ref; sudo zypper -n up', timeout => 300);
-
     record_info('Target Version', $target_version);
 
     sleep 90;    # wait for a bit for zypper to be available
 
-    record_info('installs SLE15-Migration and suse-migration-sle15-activation');
+    #Refresh and Update
+    $instance->ssh_assert_script_run( 'sudo zypper ref; sudo zypper -n up',
+        timeout => 300 );
+
+    #Download unpublished DMS package and install rpm
+    my $url              = get_var('PUBLIC_CLOUD_DMS_IMAGE_LOCATION');
+    my $package          = get_var('PUBLIC_CLOUD_DMS_PACKAGE');
+    my $pc_url           = get_var('PUBLIC_CLOUD_DMS_PC_LOCATION');
+    my $pc_pkg           = get_var('PUBLIC_CLOUD_DMS_PC_PACKAGE');
+    my $act_url          = get_var('PUBLIC_CLOUD_DMS_ACT_LOCATION');
+    my $act_pkg          = get_var('PUBLIC_CLOUD_DMS_ACT_PACKAGE');
+    my $dms_repo_key_url = get_var('PUBLIC_CLOUD_DMS_REPO_KEY_URL');
+    my $dms_repo_key     = get_var('PUBLIC_CLOUD_DMS_REPO_KEY');
+    my $dms_rpm          = "$url" . "$package";
+    my $act_rpm          = 'N/A';
+    my $pc_rpm           = 'N/A';
+    my ( $v1, $dms_rpm_name ) = split /\//, $package;
+    my $act_out = `curl -s $act_url`;
+    $act_rpm = $1 if $act_out =~ /($act_pkg.*?rpm)/g;
+    my $pc_out = `curl -s $pc_url`;
+    $pc_rpm = $1 if $pc_out =~ /($pc_pkg.*?rpm)/g;
```

Comment:
isn't this duplication of what you already has in `PUBLIC_CLOUD_DMS_PC_PACKAGE` ?

Reply:
that does not give the complete rpm name just partial.. it should be like this
suse-migration-pre-checks-2.0.36-1.1.noarch.rpm